### PR TITLE
[connector/count] Allow counting by attributes

### LINKED ISSUE #18893: Count connector - Add dimensions based on OTTL
### Component(s)

connector/count

### Is your feature request related to a problem? Please describe.

Currently metrics output using the count connector may lose context that was available in the source trace/metric/log.

Being able to add attributes from the source trace/metric/log using OTTL would help retain that context.

For example logs containing info about batch processes are counted, during the count important context about which batch and which processors ran the batch job is lost. Being able to add that context to the output metric would be ideal.

### Describe the solution you'd like

The ability to add dimensions to the count metric created by the count processor based off of OTTL (similar to the `statements` used in the transform processor).

Example to add an attribute to `batch.count.metric`  called `new.attribute` based on the attribute `source.attribute` from a source log:
```
connectors:
  count:
    logs:
      batch.count.metric:
        conditions: 
          - attributes["batch.name"] == "*payment.batch*"
        statements:
          - set(attributes["new.attribute"], attributes["source.attribute"]) where attributes["source.attribute"] != nil
```

### Describe alternatives you've considered

This seems like the most straightforward solution that matches other patterns and prior art in the collector contrib.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #18893

### CODE REVIEW

File: connector/countconnector/config.go [hunk 1]
```diff
@@ -113,6 +131,18 @@ func (c *Config) Validate() error {
 		if _, err = parseConditions(parser, info.Conditions); err != nil {
 			return fmt.Errorf("logs condition: metric %q: %w", name, err)
 		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes: metric %q: %w", name, err)
```
Comment (atoulme) [inline connector/countconnector/config.go:135]:
```suggestion
			return fmt.Errorf("logs attributes: metric %q: %w", name, err)
```

File: connector/countconnector/config.go [hunk 2]
```diff
@@ -101,6 +116,9 @@ func (c *Config) Validate() error {
 		if _, err = parseConditions(parser, info.Conditions); err != nil {
 			return fmt.Errorf("datapoints condition: metric %q: %w", name, err)
 		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes: metric %q: %w", name, err)
```
Comment (atoulme) [inline connector/countconnector/config.go:120]:
```suggestion
			return fmt.Errorf("metrics attributes: metric %q: %w", name, err)
```

File: connector/countconnector/config.go [hunk 3]
```diff
@@ -76,6 +85,9 @@ func (c *Config) Validate() error {
 		if _, err = parseConditions(parser, info.Conditions); err != nil {
 			return fmt.Errorf("spanevents condition: metric %q: %w", name, err)
 		}
+		if err := info.validateAttributes(); err != nil {
+			return fmt.Errorf("spans attributes: metric %q: %w", name, err)
```
Comment (atoulme) [inline connector/countconnector/config.go:89]:
```suggestion
			return fmt.Errorf("spanevents attributes: metric %q: %w", name, err)
```

**Review by atoulme — COMMENTED** [review]
First pass - saw some error messages that need to be differentiated per signal type.

File: connector/countconnector/config.go (hunk 3 above)
Comment (djaglowski) [reply connector/countconnector/config.go:89]: Thanks, nice catch.

**Review by atoulme — APPROVED** [review]

**Pull request merged**